### PR TITLE
[CSL 1498] Add missing query parameters (Search)

### PR DESCRIPTION
**Pull request by jjl014**
### Updates:
* Added the following missing query params: `qs`, `now`, `pre_filter_expression`, `offset`
* All tests pass

### CODE REVIEW

**Comment by jjl014** [comment]
Updated request URL creation to only include the `page` parameter if it's value is 1 (similarly for `offset` if it's 0)

File: constructorio-client/src/main/java/io/constructor/client/ConstructorIO.java [hunk 1]
```diff
@@ -1329,6 +1358,20 @@ protected Builder makeUserRequestBuilder(UserInfo info) {
         return builder;
     }
 
+    /**
+     * Creates a builder for an end user request
+     *
+     * @param info user information if available
+     * @return Request Builder
+     */
+    protected Builder makeUserRequestBuilder(UserInfo info, Boolean authorizedRequest) {
```
Comment (esezen) [inline constructorio-client/src/main/java/io/constructor/client/ConstructorIO.java:1367]: Nice

File: constructorio-client/src/main/java/io/constructor/client/ConstructorIO.java [hunk 2]
```diff
@@ -558,18 +557,48 @@ protected Request createSearchRequest(SearchRequest req, UserInfo userInfo) thro
...
+                    .build();
+            }
+
+            if (req.getOffset() != 0) {
+                url = url.newBuilder()
+                    .addQueryParameter("offset", String.valueOf(req.getOffset()))
+                    .build();
+            }
+
+            Request request = this.makeUserRequestBuilder(userInfo, true)
```
Comment (esezen) [inline constructorio-client/src/main/java/io/constructor/client/ConstructorIO.java:601]: Should this be "If now is passed, add auth. If not, do not" or do we always want to pass auth?

**Review by esezen — APPROVED** [review]
Tests pass. LGTM! Left one small comment.

File: constructorio-client/src/main/java/io/constructor/client/ConstructorIO.java (hunk 2 above)
Comment (jjl014) [reply constructorio-client/src/main/java/io/constructor/client/ConstructorIO.java:601]: Ooo, good point. Will update!

File: constructorio-client/src/main/java/io/constructor/client/ConstructorIO.java [hunk 3]
```diff
@@ -1329,6 +1364,22 @@ protected Builder makeUserRequestBuilder(UserInfo info) {
         return builder;
     }
 
+    /**
+     * Creates a builder for an end user request
+     *
+     * @param info user information if available
+     * @return Request Builder
+     */
+    protected Builder makeUserRequestBuilder(UserInfo info, Boolean authorizedRequest) {
```
Comment (jjl014) [inline constructorio-client/src/main/java/io/constructor/client/ConstructorIO.java:1373]: Missing `authorizedRequest` param in comments

File: constructorio-client/src/main/java/io/constructor/client/SearchRequest.java [hunk 4]
```diff
@@ -218,13 +226,69 @@ public List<String> getHiddenFacets() {
...
+     * @return the qs parameter
+    */
+    public String getQsParam() {
+      return qsParam;
+    }
+
+    /**
+     * @param now a unix epoch timestamp used to emulate "past" and "future" requests
+     */
+    public void setNow(String now) {
```
Comment (jjl014) [inline constructorio-client/src/main/java/io/constructor/client/SearchRequest.java:270]: Add Please refer to https://docs.constructor.io/rest_api/search/queries/

**Pull request merged**